src/lib/stitching/causality/stitching.ts

```diff
@@ -209,29 +212,94 @@ export const causalityStitchingEnvironment = ({
+              first,
+              ...rest,
+            }
+            const params = convertConnectionArgsToGravityArgs(connectionOptions)
+            console.warn({ params, connectionOptions })
+            // @ts-expect-error FIXME: Make `page` an optional parameter on `params`
+            delete params.page
+
+            const { body, headers } = await saleArtworksAllLoader(params)
+            const watchedSaleArtworks: any[] = body
+            const totalCount = parseInt(headers["x-total-count"] || "0", 10)
+
+            // fetch lot states from causality
+            // - We must explicitly query for fields here
+            // and perform money transformations below to approximate
+            // the lotState interface.
+            const lotStatesResponse = await causalityLoader({
+              query: gql`
+                query WatchedLotsQuery($ids: [ID!]!) {
+                  lots(ids: $ids) {
+                    internalID
+                    bidCount
+                    reserveStatus
+                    sellingPriceCents
+                    onlineAskingPriceCents
+                    floorSellingPriceCents
+                    soldStatus
+                  }
+                }
+              `,
+              variables: {
+                ids: watchedSaleArtworks.map((sa) => sa._id),
               },
-            }))
+            }).then((res: FetchResponse) => res.json())
+
+            const lots: any[] = lotStatesResponse?.data?.lots
+
+            if (!lots) {
+              return null
+            }
+
+            // create our final list of `Lot` nodes combining data from both services
+            const nodes = watchedSaleArtworks.reduce((acc, saleArtwork) => {
+              const lot = lots.find((l) => l.internalID === saleArtwork._id)
+              if (!lot) {
+                console.warn(
+                  `lot state for ${saleArtwork._id} not found - skipping`
+                )
+                return acc
+              }
+
+              const {
+                onlineAskingPriceCents,
+                sellingPriceCents,
+                floorSellingPriceCents,
+              } = lot
+
+              return [
+                ...acc,
+                {
+                  saleArtwork,
+                  lot: {
+                    ...lot,
+                    onlineAskingPrice: resolveLotCentsFieldToMoney(
+                      onlineAskingPriceCents
+                    ),
+                    sellingPrice: resolveLotCentsFieldToMoney(
+                      sellingPriceCents
+                    ),
+                    floorSellingPrice: resolveLotCentsFieldToMoney(
+                      floorSellingPriceCents
+                    ),
+                  },
+                },
+              ]
+            }, [])
 
             return {
-              totalCount: watchedSaleArtworks.length,
-              edges: nodes.map((node) => ({ node })),
+              totalCount,
```

Comment:
This minimal relay connection interface may behave unpredictably due to the lots + total count being derived from gravity but filtered based on available causality lot states. I'm not sure if there are any other gotchas to worry about here.